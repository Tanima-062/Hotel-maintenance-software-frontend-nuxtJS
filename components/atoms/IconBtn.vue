<template>
  <button
    type="button"
    class="flex h-fit-content items-center justify-center"
    :class="styles"
    :disabled="disabled"
    @click="click"
  >
    <svg
      class="w-7 h-7 fill-current"
      :class="`${color} hover:${hoverColor}`"
      :style="`width: ${iconWidth}px; height: ${iconHeight}px`"
      viewBox="0 0 24 24"
    >
      <path :d="icon" />
    </svg>
  </button>
</template>
<script lang="ts">
import { Vue, Component, Prop, Emit } from 'nuxt-property-decorator'

@Component
export default class extends Vue {
  @Prop({ default: '' })
  icon!: string

  @Prop({ default: 'text-white' })
  color!: string

  @Prop({ default: 'text-white' })
  hoverColor!: string

  @Prop({ default: false })
  disabled!: boolean

  @Prop({ default: false })
  round!: boolean

  @Prop({ default: '' })
  bgColor!: string

  @Prop({ default: '' })
  padding!: string

  @Prop({ default: 24 })
  iconWidth!: number

  @Prop({ default: 24 })
  iconHeight!: number

  @Prop({ default: '' })
  extraClass!: string

  private isAnim: boolean = false
  private interval: number | undefined

  get styles(): string {
    const classList = []

    classList.push(this.isAnim ? 'btn--click' : 'btn')
    if (this.disabled) {
      classList.push('cursor-not-allowed')
    }
    if (this.round) {
      classList.push('rounded-full')
    } else if (!this.extraClass.includes('rounded-sm')) {
      classList.push('rounded-md')
    }
    classList.push(this.bgColor ? `${this.bgColor} opacity-75` : 'opacity-85')
    if (this.padding) {
      classList.push(`${this.padding}`)
    }
    if (this.extraClass) {
      classList.push(this.extraClass)
    }
    return classList.join(' ')
  }

  private doAnim(): void {
    if (this.isAnim || this.interval) {
      return
    }
    this.isAnim = true
    this.interval = window.setInterval(() => {
      this.isAnim = false
      clearInterval(this.interval)
      this.interval = undefined
    }, 130)
  }

  @Emit()
  private click(e: Event): Event {
    this.doAnim()
    return e
  }
}
</script>
<style lang="sass" scoped>
.btn
  transition: opacity .1s ease-in
  &:hover
    transition: opacity .2s cubic-bezier(.4,0,.6,1)
    opacity: 1
  &--click
    transition: opacity .05s ease-out
    opacity: .7
.h-fit-content
  height: fit-content
</style>
