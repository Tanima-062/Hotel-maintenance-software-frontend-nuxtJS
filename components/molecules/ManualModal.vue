<template>
  <transition name="fade-modal">
    <div class="fixed z-40">
      <div class="opacity-25 fixed inset-0 z-10 bg-black"></div>
      <div
        id="toTopId"
        class="inset-0 fixed overflow-auto h-screen py-18 top-0 left-0 z-40 flex"
        @click.self="closeModal"
      >
        <div class="relative manual-w m-auto">
          <div
            class="border-0 p-5 rounded-lg shadow-lg relative flex flex-col z-20 w-full bg-white"
          >
            <div class="flex items-start justify-between mb-2 rounded-t">
              <h3
                class="text-base font-bold text-black-dark whitespace-no-wrap"
              >
                {{ manualData.title }}
              </h3>
              <IconBtn
                :icon="icon.closeIcon"
                hover-color="text-black-dark"
                color="text-black-dark"
                class="text-black-dark h-6 w-6 text-xl block"
                @click="closeModal"
              />
            </div>
            <div class="">
              <div v-for="(image, index) in manualData.imageList" :key="index">
                <img :src="image" />
                <div
                  v-if="
                    manualData.separateOption === 'arrow' &&
                    index < manualData.imageList.length - 1
                  "
                  class="w-full p-2 flex justify-center flex-col"
                >
                  <img class="w-6 m-auto" :src="arrowImages" />
                  <img class="w-6 m-auto" :src="arrowImages" />
                </div>
                <div
                  v-if="
                    manualData.separateOption === 'dashed' &&
                    index < manualData.imageList.length - 1
                  "
                  class="border-b border-inputLength border-dashed w-full my-4"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </transition>
</template>
<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator'
import { mdiClose } from '@mdi/js'
import { Global } from '~/types'

@Component({
  components: {
    IconBtn: () => import('~/components/atoms/IconBtn.vue'),
  },
})
export default class ManualModal extends Vue {
  @Prop({ required: true })
  manualData!: Global.ManualItem

  @Prop({ required: true })
  closeManualModal!: Function

  private icon: { [key: string]: string } = {
    closeIcon: mdiClose,
  }

  // Arrow Icon Image
  private arrowImages = '/images/guide-arrow.svg'

  private closeModal(): void {
    this.scrollToTopModal()
    this.closeManualModal()
  }

  private scrollToTopModal(): void {
    const scroll = document.getElementById('toTopId')
    if (!scroll) return
    scroll.scrollTop = 0
  }
}
</script>
<style lang="sass" scoped>
.manual-w
  width: 53rem
.fade-modal-enter-active, .fade-modal-leave-active
  transition: all 0.4s
.fade-modal-enter, .fade-modal-leave-to
  opacity: 0
</style>
