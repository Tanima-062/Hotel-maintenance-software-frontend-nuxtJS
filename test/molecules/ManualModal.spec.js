import { mount } from '@vue/test-utils'
import ManualModal from '@/components/molecules/ManualModal.vue'
import IconBtn from '@/components/atoms/IconBtn.vue'

describe('Molecules ManualModal Check', () => {
  test('ManualModal Test', () => {
    const props = {
      manualData: {
        title: 'test',
        separateOption: 'arrow',
        imageList: [
          '/images/hotel-guide01-1.jpg',
          '/images/hotel-guide01-2.jpg',
        ],
      },
      closeManualModal: () => {},
    }
    const wrapper = mount(ManualModal, { propsData: props, stubs: { IconBtn } })
    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })
})
