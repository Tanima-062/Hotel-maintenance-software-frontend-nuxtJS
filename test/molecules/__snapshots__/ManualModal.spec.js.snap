// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules ManualModal Check ManualModal Test 1`] = `
<transition-stub
  name="fade-modal"
>
  <div
    class="fixed z-40"
  >
    <div
      class="opacity-25 fixed inset-0 z-10 bg-black"
    />
     
    <div
      class="inset-0 fixed overflow-auto h-screen py-18 top-0 left-0 z-40 flex"
      id="toTopId"
    >
      <div
        class="relative manual-w m-auto"
      >
        <div
          class="border-0 p-5 rounded-lg shadow-lg relative flex flex-col z-20 w-full bg-white"
        >
          <div
            class="flex items-start justify-between mb-2 rounded-t"
          >
            <h3
              class="text-base font-bold text-black-dark whitespace-no-wrap"
            >
              
              test
            
            </h3>
             
            <button
              class="flex h-fit-content items-center justify-center text-black-dark h-6 w-6 text-xl block btn rounded-md opacity-85"
              type="button"
            >
              <svg
                class="w-7 h-7 fill-current text-black-dark hover:text-black-dark"
                style="width: 24px; height: 24px;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
                />
              </svg>
            </button>
          </div>
           
          <div>
            <div>
              <img
                src="/images/hotel-guide01-1.jpg"
              />
               
              <div
                class="w-full p-2 flex justify-center flex-col"
              >
                <img
                  class="w-6 m-auto"
                  src="/images/guide-arrow.svg"
                />
                 
                <img
                  class="w-6 m-auto"
                  src="/images/guide-arrow.svg"
                />
              </div>
               
              <!---->
            </div>
            <div>
              <img
                src="/images/hotel-guide01-2.jpg"
              />
               
              <!---->
               
              <!---->
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</transition-stub>
`;
